src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -1121,19 +1121,15 @@ winrt::fire_and_forget Pane::_playBellSound(winrt::Windows::Foundation::Uri uri)
     co_await wil::resume_foreground(_root.Dispatcher());
     if (auto pane{ weakThis.lock() })
     {
-        // BODGY
-        // GH#12258: We learned that if you leave the MediaPlayer open, and
-        // press the media keys (like play/pause), then the OS will _replay the
-        // bell_. So we have to re-create the MediaPlayer each time we want to
-        // play the bell, to make sure a subsequent play doesn't come through
-        // and reactivate the old one.
-
-        if (!_bellPlayer)
+        if (!_bellPlayerCreated)

```

Comment:
Does the boolean here prevent us from eating an exception on line 1130 every time there's a bell on non-media builds of Windows?

Reply:
Yep exactly! Although I suspect it's more of a theoretical thing (since most won't enable it if it doesn't work), I feel like this is more "polished".